packages/fhir-converter/src/lambdas/ccda-to-fhir.js

```diff
@@ -0,0 +1,192 @@
+var path = require("path");
+var fs = require("fs");
+var Promise = require("promise");
+var compileCache = require("memory-cache");
+var constants = require("./lib/constants/constants");
+var errorCodes = require("./lib/error/error").errorCodes;
+var errorMessage = require("./lib/error/error").errorMessage;
+var HandlebarsConverter = require("./lib/handlebars-converter/handlebars-converter");
+var dataHandlerFactory = require("./lib/dataHandler/dataHandlerFactory");
+var {
+  extractEncounterTimePeriod,
+  getEncompassingEncounterId,
+} = require("./lib/inputProcessor/dateProcessor");
+
+const { createNamespace } = require("cls-hooked");
+var session = createNamespace(constants.CLS_NAMESPACE);
+
+var rebuildCache = true;
+
+function GetHandlebarsInstance(dataTypeHandler, templatesMap) {
+  // New instance should be created when using templatesMap
+  let needToUseMap =
+    templatesMap && Object.entries(templatesMap).length > 0 && templatesMap.constructor === Object;
+  var instance = HandlebarsConverter.instance(
+    needToUseMap ? true : rebuildCache,
+    dataTypeHandler,
+    path.join(constants.BASE_TEMPLATE_FILES_LOCATION, dataTypeHandler.dataType),
+    templatesMap
+  );
+  rebuildCache = needToUseMap ? true : false; // New instance should be created also after templatesMap usage
+
+  return instance;
```

Comment:
a wee bit of code copy pasta, but not things we'll modify generally

will refactor this into common js files later